embassy-rp/src/dma.rs

```diff
@@ -297,3 +303,11 @@ channel!(DMA_CH8, 8);
 channel!(DMA_CH9, 9);
 channel!(DMA_CH10, 10);
 channel!(DMA_CH11, 11);
+#[cfg(feature = "rp235x")]
```

Comment:
add a cfg to `CHANNEL_COUNT`, or the higher newer channels will panic due to out of bounds.